cli/src/cmd/forge/doc.rs

```diff
@@ -0,0 +1,138 @@
+use crate::cmd::Cmd;
+use clap::{Parser, ValueHint};
+use forge_doc::{SolidityDoc, SolidityDocPartElement};
+use forge_fmt::Visitable;
+use foundry_config::{find_project_root_path, load_config_with_root};
+use rayon::prelude::*;
+use solang_parser::doccomment::DocComment;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+#[derive(Debug, Clone, Parser)]
+pub struct DocArgs {
+    #[clap(
+        help = "The project's root path.",
+        long_help = "The project's root path. By default, this is the root directory of the current Git repository, or the current working directory.",
+        long,
+        value_hint = ValueHint::DirPath,
+        value_name = "PATH"
+    )]
+    root: Option<PathBuf>,
+}
+
+impl DocArgs {
+    fn format_comments(&self, comments: &Vec<DocComment>) -> String {
+        comments
+            .iter()
+            .map(|c| match c {
+                DocComment::Line { comment } => comment.value.to_owned(),
+                DocComment::Block { comments } => comments
+                    .iter()
+                    .map(|comment| comment.value.to_owned())
+                    .collect::<Vec<_>>()
+                    .join("\n"),
+            })
+            .collect::<Vec<_>>()
+            .join("\n")
+    }
+}
+
+impl Cmd for DocArgs {
+    type Output = ();
+
+    fn run(self) -> eyre::Result<Self::Output> {
+        let config = load_config_with_root(self.root.clone());
+        let paths: Vec<PathBuf> = config.project_paths().input_files();
+        let docs = paths
+            .par_iter()
+            .enumerate()
+            .map(|(i, path)| {
+                let source =  fs::read_to_string(&path)?;
+                let (mut source_unit, comments) = solang_parser::parse(&source, i)
+                    .map_err(|diags| eyre::eyre!(
+                            "Failed to parse Solidity code for {}. Leaving source unchanged.\nDebug info: {:?}",
+                            path.display(),
+                            diags
+                        ))?;
+                let mut doc = SolidityDoc::new(comments);
+                source_unit.visit(&mut doc)?;
+
+                Ok((path.clone(), doc.parts))
+            })
+            .collect::<eyre::Result<Vec<_>>>()?;
+
+        let out_dir = self.root.as_ref().unwrap_or(&find_project_root_path()?).join("docs");
+        fs::create_dir_all(&out_dir.join("src"))?;
```

Comment:
Should this read the value from `foundry.toml` instead of hardcoding src?

Reply:
yup, as well as the title & description for the docs. i put it out as a POC in order to receive comments regarding the implementation & output format. based on the comments, i will change it be more rustdoc style like